partiql-lang/src/main/kotlin/org/partiql/lang/graph/SimpleGraph.kt

```diff
@@ -32,4 +32,57 @@ class SimpleGraph(
     companion object {
         val empty: Graph = SimpleGraph(emptyList(), emptyList(), emptyList())
     }
+
+    private fun labelsMatchSpec(labels: Set<String>, spec: LabelSpec): Boolean =
+        when (spec) {
+            LabelSpec.Whatever -> true
+            is LabelSpec.OneOf -> labels.contains(spec.name)
+        }
+
+    private fun Graph.Elem.matches(labelSpec: LabelSpec): Boolean =
+        labelsMatchSpec(this.labels, labelSpec)
+
+    override fun getNodes(spec: LabelSpec): List<Node> {
```

Comment:
maybe a name that implies searching/filtering; `getNodes` implied returning all nodes to me.

Reply:
`scanNodes` ?